### ISSUE TITLE
Voiceover is not announced "item added or updated" when user is adding or editing a new Recipe or existing Recipe: A11y_.NET Core_Recipes App My Recipes Edit_Edit Recipe_Screen reader.

### ISSUE THREAD

**Issue by keerthiduvvuri** [dotnet/maui#23001]
### GIT Hub Tags:
#AI Limited; #A11yTCS; #A11ySev2; #.NET Core; #E2E-.NET Core-IOS-MAUI-JUNE2024; #Voiceover; #IOS; #WCAG1.3.1; #Info and Relationship; #FTP; #A11yMAS; #ScreenReader; #Elemet: Screenreader;

### Environment and OS details:
IOS Version 17

### Reproduction Steps

1. Install and open ".NET MAUI" app.
2. Swipe to "My Recipes" and double tap.
3. Swipe to any existing recipe and double tap.
4. "Recipe" Screen will open.
5. Swipe to "Edit" button and double tap.
6. Verify all the controls in the "Edit recipe" are accessible and MAS Compliant.
7. Enter text in all entry fields.
8. Save recipe.
9. Observe that Voiceover is not announced "item added or updated" when user is adding or editing a new Recipe or existing Recipe.

### Actual behavior:
Voiceover is not announced "item added or updated" when user is adding or editing a new Recipe or existing Recipe.

### Expected behavior
Voiceover should be announced "item added or updated" when user is adding or editing a new Recipe or existing Recipe.

### User Impact:
Screen reader users will face difficulty if Voiceover is not announced "item added or updated" when user is adding or editing a new Recipe or existing Recipe.

### Attachments:

https://github.com/dotnet/maui/assets/98728920/6281b667-3e49-4afe-b736-b2dcf7087716

https://github.com/dotnet/maui/assets/98728920/11a89429-4e98-4980-92e1-d4be256038fc

**Comment by keerthiduvvuri** [comment]
@RoiChen001 sorry i have changed back to IOS, the bug is in IOS platform.

**Comment by Yash14j** [comment]
GithubTags:#Rev:yaja;

**Comment by tj-devel709** [comment]
I have verified this behaves as stated above.

I wonder if the issue is the timing with the `await Shell.Current.GoToAsync("..");` right after the `SemanticScreenReader.Announce(RecipeName + " recipe added.");` call. 

If I remove the GoToAsync call, the full announcement is made. 

Not sure how manageable this would be, but perhaps adding an async Announce method could be worth trying [here](https://github.com/dotnet/maui/blob/recipes/src/Essentials/src/SemanticScreenReader/SemanticScreenReader.ios.cs):

```cs
namespace Microsoft.Maui.Essentials
{
    public static partial class SemanticScreenReader
    {
        public static async Task Announce(string text)
        {
            if (!UIAccessibility.IsVoiceOverRunning)
                return;

            var tcs = new TaskCompletionSource<bool>();

            MainThread.BeginInvokeOnMainThread(async () =>
            {
                await Task.Delay(100);
                UIAccessibility.PostNotification(UIAccessibilityPostNotification.Announcement, new NSString(text));
                tcs.SetResult(true);
            });

            await tcs.Task;
        }
    }
}
```

Worth noting that there were timing issues with iOS Screenreader here: https://github.com/dotnet/maui/issues/3736

**Comment by tj-devel709** [comment]
There is also an interesting workaround here: https://stackoverflow.com/a/59329208

**Comment by keerthiduvvuri** [comment]
Verified in latest environment, Issue is Fixed, Hence closing the bug.
Added attachment for reference.

#Regressed:09-11-24; #Closed;


https://github.com/user-attachments/assets/c444e200-38ab-4d57-9436-b12001f14365



https://github.com/user-attachments/assets/81eb77b0-be0b-4780-b742-a37ecc0a3b88
